test/runtime_tests.cc

```diff
@@ -1013,5 +1015,59 @@ namespace detail {
 		dry_run_with_nodes(nodes);
 	}
 
+	TEST_CASE_METHOD(test_utils::mpi_fixture, "Config reads environment variables correctly", "[env-vars]") {
+		std::unordered_map<std::string, std::string> valid_test_env_vars{
+		    {"CELERITY_LOG_LEVEL", "debug"},
+		    {"CELERITY_GRAPH_PRINT_MAX_VERTS", "1"},
+		    {"CELERITY_DEVICES", "0 0"},
+		    {"CELERITY_PROFILE_KERNEL", "1"},
+		    {"CELERITY_DRY_RUN_NODES", "4"},
+		};
+
+		const auto test_env = env::scoped_test_environment(valid_test_env_vars);
+		auto cfg = config(nullptr, nullptr);
+
+		REQUIRE(spdlog::get_level() == spdlog::level::debug);
+		REQUIRE(cfg.get_graph_print_max_verts() == 1);
+		REQUIRE(config_testspy::get_device_list(cfg) == std::vector<size_t>{0, 0});
+		const auto has_prof = cfg.get_enable_device_profiling();
+		REQUIRE(has_prof.has_value());
+		REQUIRE((*has_prof) == true);
+		REQUIRE(cfg.get_dry_run_nodes() == 4);
+	}
+
+	TEST_CASE_METHOD(test_utils::mpi_fixture, "Config reports incorrect environment varibles", "[env-vars]") {
+		const std::string error_string{"Failed to parse/validate environment variables."};
+		{
+			std::unordered_map<std::string, std::string> invalid_test_env_var{{"CELERITY_LOG_LEVEL", "a"}};
+			const auto test_env = env::scoped_test_environment(invalid_test_env_var);
+			CHECK_THROWS_WITH((celerity::detail::config(nullptr, nullptr)), error_string);
+		}
+
+		{
+			std::unordered_map<std::string, std::string> invalid_test_env_var{{"CELERITY_GRAPH_PRINT_MAX_VERTS", "a"}};
+			const auto test_env = env::scoped_test_environment(invalid_test_env_var);
+			CHECK_THROWS_WITH((celerity::detail::config(nullptr, nullptr)), error_string);
+		}
+
+		{
+			std::unordered_map<std::string, std::string> invalid_test_env_var{{"CELERITY_DEVICES", "a"}};
+			const auto test_env = env::scoped_test_environment(invalid_test_env_var);
+			CHECK_THROWS_WITH((celerity::detail::config(nullptr, nullptr)), error_string);
+		}
+
+		{
+			std::unordered_map<std::string, std::string> invalid_test_env_var{{"CELERITY_DRY_RUN_NODES", "a"}};
+			const auto test_env = env::scoped_test_environment(invalid_test_env_var);
+			CHECK_THROWS_WITH((celerity::detail::config(nullptr, nullptr)), error_string);
+		}
+
+		{
+			std::unordered_map<std::string, std::string> invalid_test_env_var{{"CELERITY_PROFILE_KERNEL", "a"}};
+			const auto test_env = env::scoped_test_environment(invalid_test_env_var);
+			CHECK_THROWS_WITH((celerity::detail::config(nullptr, nullptr)), error_string);
+		}
```

Comment:
The two deprecated environment variables `CELERITY_PROFILE_OCL` and `CELERITY_FORCE_WG` should also be tested.